src/sign_tx_hash.c

```diff
@@ -6,14 +6,197 @@
+    bool has_data = strlen(tx_context.data) > 0;
+    bool has_esdt_transfer_prefix = false;
+    bool has_registered_esdt_identifier = false;
+    bool next_char_after_identifier_is_at_separator = false;
+    bool same_chainid = false;
+
+    if (!esdt_info.valid) {
+        return false;
+    }
+
+    if (has_data && identifier_len_valid) {
+        has_esdt_transfer_prefix = strncmp(tx_context.data + DATA_SIZE_LEN - 1,
+                                           ESDT_TRANSFER_PREFIX,
+                                           ESDT_TRANSFER_PREFIX_LENGTH) == 0;
+        has_registered_esdt_identifier =
+            strncmp(tx_context.data + DATA_SIZE_LEN - 1 + ESDT_TRANSFER_PREFIX_LENGTH,
+                    esdt_info.identifier,
+                    esdt_info.identifier_len) == 0;
+
+        size_t identifier_end_position =
+            DATA_SIZE_LEN - 1 + ESDT_TRANSFER_PREFIX_LENGTH + esdt_info.identifier_len;
+        next_char_after_identifier_is_at_separator =
+            (strlen(tx_context.data) > identifier_end_position) &&
+            (tx_context.data[identifier_end_position] == '@');
+
+        same_chainid = strncmp(tx_context.chain_id, esdt_info.chain_id, MAX_CHAINID_LEN) == 0;
+    }
+
+    return has_esdt_transfer_prefix && has_registered_esdt_identifier &&
+           next_char_after_identifier_is_at_separator && same_chainid;
+}
+
+#if defined(TARGET_STAX)
+
+static void start_review(void);
+static void ui_sign_tx_hash_nbgl(void);
+static void rejectUseCaseChoice(void);
+static nbgl_layoutTagValueList_t layout;
+static nbgl_layoutTagValue_t infos[5];  // 5 info max for ESDT and 5 info max for EGLD
+
+static const nbgl_pageInfoLongPress_t review_final_long_press = {
+    .text = "Sign transaction on\nMultiversX network?",
+    .icon = &C_icon_multiversx_logo_64x64,
+    .longPressText = "Hold to sign",
+    .longPressToken = 0,
+    .tuneId = TUNE_TAP_CASUAL,
+};
+
+static void review_final_callback(bool confirmed) {
+    if (confirmed) {
+        int tx = set_result_signature();
+        io_seproxyhal_io_heartbeat();
+        send_response(tx, true, false);
+        nbgl_useCaseStatus("TRANSACTION\nSIGNED", true, ui_idle);
+    } else {
+        rejectUseCaseChoice();
+    }
+}
+
+static void rejectChoice(void) {
+    send_response(0, false, false);
+    nbgl_useCaseStatus("Transaction\nrejected", false, ui_idle);
+}
+
+static void rejectUseCaseChoice(void) {
+    nbgl_useCaseConfirm("Reject transaction?",
+                        NULL,
+                        "Yes, reject",
+                        "Go back to transaction",
+                        rejectChoice);
+}
+
+static void start_review(void) {
+    uint8_t step = 0;
+
+    if (should_display_esdt_flow) {
+        infos[step].item = "Token", infos[step].value = esdt_info.ticker, ++step;
+        infos[step].item = "Receiver", infos[step].value = tx_context.receiver, ++step;
+        infos[step].item = "Value", infos[step].value = tx_context.amount, ++step;
```

Comment:
the old flow had the order `Token -> Value -> Receiver`. is the new order desired instead? If so, the Nano version should be updated for consistency

Reply:
Also, no data here?

Reply:
No, since this is the custom flow for allowed ESDT tokens (basically, the info is extracted from `ESDTTransfer@<token>@<value>` and places into those fields)

Reply:
Oh, ok, got it, thanks

Reply:
Nice catch, I've reverted to `Token -> Value -> Receiver` for consistency